rcl/src/rcl/node.c

```diff
@@ -310,6 +317,16 @@ rcl_node_init(
   goto cleanup;
 fail:
   if (node->impl) {
+    ret = rcl_node_type_description_service_fini(node);
+    RCUTILS_LOG_ERROR_EXPRESSION_NAMED(
+      (ret != RCL_RET_OK && ret != RCL_RET_NOT_INIT), ROS_PACKAGE_NAME,
+      "Failed to fini get_type_description service for node: %i", ret);
+
+    ret = rcl_node_type_cache_fini(node);
+    RCUTILS_LOG_ERROR_EXPRESSION_NAMED(
+      (ret != RCL_RET_OK && ret != RCL_RET_NOT_INIT), ROS_PACKAGE_NAME,
+      "Failed to fini node_type_cache for node: %i", ret);
```

Comment:
Similarly, I think we only need to do this teardown if we initialized it to begin with.  This is not a huge deal, but might avoid some unrelated spam on the console when tearing down from a failure.

Reply:
These two return `RCL_NOT_INIT` if the object is not initialized, and the error printing checks for `ret != RCL_RET_OK && ret != RCL_RET_NOT_INIT` - so if the objects weren't initialized they will move on silently.

Reply:
Sorry - you're right, it will still leave an error state that will add spam in _other_ failure teardown. Will address